File: generators/mgtCephalids.py
__all__ = ['mgtCephalids']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm3', 'nm7', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
            var.put('rnd8', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
            if (var.get('i')<Js(6.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                while PyJsStrictEq(var.get('nm2').get(var.get('rnd2')),var.get('nm4').get(var.get('rnd4'))):
                    var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                while PyJsStrictEq(var.get('nm5').get(var.get('rnd6')),var.get('nm4').get(var.get('rnd4'))):
                    var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                var.put('names', (((((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm4').get(var.get('rnd4')))+var.get('nm3').get(var.get('rnd5')))+var.get('nm5').get(var.get('rnd6')))+Js(', '))+var.get('nm6').get(var.get('rnd7')))+Js(' '))+var.get('nm7').get(var.get('rnd8'))))
            else:
                var.put('names', ((var.get('nm6').get(var.get('rnd7'))+Js(' '))+var.get('nm7').get(var.get('rnd8'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('a'), Js('o'), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('')]))
var.put('nm2', Js([Js('b'), Js('d'), Js('g'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('r'), Js('t'), Js('v')]))
var.put('nm3', Js([Js('a'), Js('o'), Js('a'), Js('o'), Js('a'), Js('o'), Js('a'), Js('o'), Js('a'), Js('o'), Js('u'), Js('e'), Js('i')]))
var.put('nm4', Js([Js('c'), Js('d'), Js('dh'), Js('g'), Js('kh'), Js('l'), Js('ll'), Js('n'), Js('nn'), Js('s'), Js('ss'), Js('sh'), Js('t'), Js('th'), Js('tt'), Js('v'), Js('w'), Js('z'), Js('zh')]))
var.put('nm5', Js([Js('d'), Js('l'), Js('m'), Js('n'), Js('t'), Js('th')]))
var.put('nm6', Js([Js('Abandoned'), Js('Acclaimed'), Js('Aggressive'), Js('Agitated'), Js('Ambitious'), Js('Ancient'), Js('Bitter'), Js('Bold'), Js('Broken'), Js('Callous'), Js('Careless'), Js('Cautious'), Js('Corrupt'), Js('Corrupted'), Js('Crooked'), Js('Cruel'), Js('Dangerous'), Js('Defiant'), Js('Delirious'), Js('Deserted'), Js('Devoted'), Js('Diligent'), Js('Discrete'), Js('Dreary'), Js('Eager'), Js('Enraged'), Js('Evil'), Js('Fearless'), Js('Fickle'), Js('Forsaken'), Js('Frightening'), Js('Grand'), Js('Grave'), Js('Grim'), Js('Gruesome'), Js('Haunting'), Js('Hideous'), Js('Husky'), Js('Infernal'), Js('Juvenile'), Js('Keen'), Js('Livid'), Js('Loathsome'), Js('Lone'), Js('Lost'), Js('Lumbering'), Js('Mad'), Js('Mindless'), Js('Monstrous'), Js('Nasty'), Js('Needy'), Js('Noxious'), Js('Obedient'), Js('Petty'), Js('Powerful'), Js('Prime'), Js('Ragged'), Js('Rash'), Js('Reckless'), Js('Somber'), Js('Spiteful'), Js('Trifling'), Js('Turbulent'), Js('Unfit'), Js('Useless'), Js('Vicious'), Js('Villainous'), Js('Violent'), Js('Weak'), Js('Wicked'), Js('Wild'), Js('Woeful'), Js('Wrathful'), Js('Wretched'), Js('Zealous')]))
var.put('nm7', Js([Js('Agent'), Js('Aristocrat'), Js('Broker'), Js('Bully'), Js('Chandler'), Js('Charmer'), Js('Conjurer'), Js('Dealer'), Js('Defacer'), Js('Diviner'), Js('Escort'), Js('Explorer'), Js('Guard'), Js('Guardian'), Js('Illusionist'), Js('Informer'), Js('Lookout'), Js('Looter'), Js('Mage'), Js('Marauder'), Js('Mediator'), Js('Merchant'), Js('Monger'), Js('Negotiator'), Js('Noble'), Js('Operative'), Js('Oppressor'), Js('Outrider'), Js('Overlord'), Js('Overseer'), Js('Patrician'), Js('Patrol'), Js('Pillager'), Js('Pioneer'), Js('Plunderer'), Js('Ravager'), Js('Recruiter'), Js('Retailer'), Js('Sage'), Js('Scout'), Js('Sentinel'), Js('Sneak'), Js('Soothsayer'), Js('Spellbinder'), Js('Spoiler'), Js('Spy'), Js('Taskmaster'), Js('Trader'), Js('Tyrant'), Js('Vandal'), Js('Vanguard'), Js('Vendor'), Js('Watcher')]))
pass
pass


# Add lib to the module scope
mgtCephalids = var.to_python()

File: js/mgtCephalids.js
var nm1 = ["a","o","","","","","","","",""];
var nm2 = ["b","d","g","l","ll","m","n","r","t","v"];
var nm3 = ["a","o","a","o","a","o","a","o","a","o","u","e","i"];
var nm4 = ["c","d","dh","g","kh","l","ll","n","nn","s","ss","sh","t","th","tt","v","w","z","zh"];
var nm5 = ["d","l","m","n","t","th"];

var nm6 = ["Abandoned","Acclaimed","Aggressive","Agitated","Ambitious","Ancient","Bitter","Bold","Broken","Callous","Careless","Cautious","Corrupt","Corrupted","Crooked","Cruel","Dangerous","Defiant","Delirious","Deserted","Devoted","Diligent","Discrete","Dreary","Eager","Enraged","Evil","Fearless","Fickle","Forsaken","Frightening","Grand","Grave","Grim","Gruesome","Haunting","Hideous","Husky","Infernal","Juvenile","Keen","Livid","Loathsome","Lone","Lost","Lumbering","Mad","Mindless","Monstrous","Nasty","Needy","Noxious","Obedient","Petty","Powerful","Prime","Ragged","Rash","Reckless","Somber","Spiteful","Trifling","Turbulent","Unfit","Useless","Vicious","Villainous","Violent","Weak","Wicked","Wild","Woeful","Wrathful","Wretched","Zealous"];
var nm7 = ["Agent","Aristocrat","Broker","Bully","Chandler","Charmer","Conjurer","Dealer","Defacer","Diviner","Escort","Explorer","Guard","Guardian","Illusionist","Informer","Lookout","Looter","Mage","Marauder","Mediator","Merchant","Monger","Negotiator","Noble","Operative","Oppressor","Outrider","Overlord","Overseer","Patrician","Patrol","Pillager","Pioneer","Plunderer","Ravager","Recruiter","Retailer","Sage","Scout","Sentinel","Sneak","Soothsayer","Spellbinder","Spoiler","Spy","Taskmaster","Trader","Tyrant","Vandal","Vanguard","Vendor","Watcher"];

var br = "";

function nameGen(){
	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");

	for(i = 0; i < 10; i++){
		rnd7 = Math.floor(Math.random() * nm6.length);
		rnd8 = Math.floor(Math.random() * nm7.length);
		if(i < 6){
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			rnd3 = Math.floor(Math.random() * nm3.length);
			rnd4 = Math.floor(Math.random() * nm4.length);
			while(nm2[rnd2] === nm4[rnd4]){
				rnd4 = Math.floor(Math.random() * nm4.length);
			}
			rnd5 = Math.floor(Math.random() * nm3.length);
			rnd6 = Math.floor(Math.random() * nm5.length);
			while(nm5[rnd6] === nm4[rnd4]){
				rnd6 = Math.floor(Math.random() * nm5.length);
			}
			names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm4[rnd4] + nm3[rnd5] + nm5[rnd6] + ", " + nm6[rnd7] + " " + nm7[rnd8];
		}else{
			names = nm6[rnd7] + " " + nm7[rnd8];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}